### DISCUSSION TITLE
How to load plugin for Performer Fast Binary (PFB) format?

### DISCUSSION THREAD

**Discussion by jumpjack** [openscenegraph/OpenSceneGraph#1343] [Q&A]
I read that OSG supports ancient format "PFB" (IRIX OpenGL Performer Fast Binary) through a specific plugin "src/osgPlugins/pfb", but I am not a developer, just an user: can anybody please explain me how to load/install this plugin so as to convert a .pfg file to other formats?
I have all the libraries of OpenGL Performer, but I don't understand how to use them.
I am using Windows 11.

I also downloaded https://download.osgvisual.org/3rdParty_VS2017_v141_x64_V11_full.7z   from https://openscenegraph.github.io/OpenSceneGraphDotComBackup/OpenSceneGraph/www.openscenegraph.com/index.php/download-section/dependencies.html   but I don't know what to do with it.

**Comment by robertosfield** [comment]
I haven't been used OpenGL Performer for 24 years, the OSG's pfb loader was written in the OSG' infancy and quickly fell out of use in the early 2000's as the industry moved on and Perfomer support was dropped by SGI.  

This sketch seems apt, substitute Norwegian Blue for Performer:
https://www.youtube.com/watch?v=vZw35VUBdzo

So the question has to be, why on earth would you want to be trying to use .pfb files 2 decades after it had any relevance.

**Comment by jumpjack** [comment]
Because .pfb file is the only format used by NASA to [store mission data for martian rovers Spirit and Opportunity](https://planetarydata.jpl.nasa.gov/img/data/mer/spirit/mer2mw_0xxx/data/navcam/site0137/): there are dozens of such 3d files on NASA server, which nobody in the world can currently access anymore! I am trying to make them accessible again.

You can find here my efforts:

https://github.com/jumpjack/VST-converter

I started with VST files, which at least have a public format, but I can't go further than extracting pointcloud, then I found that probably a PFB file contains the 3d mesh corresponding to multiple VST files all merged  together.

In the meantime, I found that what I need for OSG to open pfb files is a plugin named "**osgdb_pfb.dll"** , but I can't find it anywhere.
I found the [source code and demo executable](http://windowsdraw.altervista.org/abandonware.html) of OpenGL Performer for Windows, but I am not a C programmer so I have no idea about how to use this stuff for compiling the plugin, any suggestion?

Or maybe it already exists buried somewhere  in  old versions of OSG repo?

**Comment by robertosfield** [comment]
I don't know if anyone attempted to build the OSG with Performer in order to create the pfb loader, let alone publish binaries for it.

If it's possible to still build against Performer then it might be possible to build the OSG's pfb loader, but this is really something a developer would need to do.  I'm not an active OSG user, all my time is now spent on the VSG, and just an occasional Windows user with little expertise on the platform, and am very busy with my VSG commitments so I can't personally help your out other than point you possibly helpful direction.

Your best bet would probably be to find a friendly OSG user that has a bit of Performance knowledge, and that fancies helping breath life into this old data.

**Comment by LaurensVoerman** [comment]
Hi Jumpjack,

I do have a working pfb loader, but it runs only on windows XP. Runs fine in a Hyper-V VM, and I have just tested one of the Nasa files. Of course, those have problems too, as the texture filenames ended up in lowercase, while they must be in uppercase (with lowercase extension) for this .pfb file. 
in short:
You can get my binaries (on google drive [here](https://drive.google.com/file/d/1GpOJW5vuc_igQWx7xZQxzCiNWQXmzGwd/view?usp=sharing); 178Mb ) , but they will not run on windows 11.

Regards, Laurens.

**Reply by LaurensVoerman** [reply]
Updated version (behind the same link) works on windows 10.

**Reply by jumpjack** [reply]
Very good job! Thank you very much. Works fine on Windows 11.
